entrypoint.sh

```diff
@@ -67,30 +86,46 @@ echo "========================================================================"
 
 for ((i=0;i<20;i++))
 do
-    DB_CONNECTABLE=$(mysql -u"$DB_USER" -p"$DB_PASS" -h"$DB_HOST" -P"$DB_PORT" -e 'status' >/dev/null 2>&1; echo "$?")
-    if [[ $DB_CONNECTABLE -eq 0 ]]; then
-        break
+    if [[ $DB_TYPE == "mysql" ]]; then
+        DB_CONNECTABLE=$(mysql -u"$DB_USER" -p"$DB_PASS" -h"$DB_HOST" -P"$DB_PORT" -e 'status' >/dev/null 2>&1; echo "$?")
+        if [[ $DB_CONNECTABLE -eq 0 ]]; then
+            break
+        fi
+    else
+        DB_CONNECTABLE=$(psql -U "$DB_USER" -h "$DB_HOST" -p "$DB_PORT" -l >/dev/null 2>&1; echo "$?")
+        if [[ $DB_CONNECTABLE -eq 0 ]]; then
+            break
+        fi
     fi
     sleep 3
 done
 
 if ! [[ $DB_CONNECTABLE -eq 0 ]]; then
-	echo "Cannot connect to MySQL"
+	echo "Cannot connect to database"
     exit $DB_CONNECTABLE
 fi
-
 ###
 
-if ! drush sql-query "SHOW DATABASES LIKE '${DB_NAME}';" > /dev/null ; then
-	run_scripts setup
-    echo "=> Done installing site!"
-	if [ $EXTRA_SETUP_SCRIPT ]; then
-		echo "=> WARNING: The usage of EXTRA_SETUP_SCRIPT is deprectated. Put your script into /scripts/post-setup.d/"
-		. $EXTRA_SETUP_SCRIPT
-		echo "=> Successfully ran extra setup script ${EXTRA_SETUP_SCRIPT}."
+if [[ $DB_TYPE == "mysql" ]]; then
+	if ! drush sql-query "SHOW DATABASES LIKE '${DB_NAME}';" > /dev/null ; then
```

Comment:
i'd prefer a query that is defined earlier depending on the database-driver and then feed to `drush sql-query`. would that be possible? if necessary one could also define a driver-based output that is expected and used for branching.